zerver/tests/test_push_notifications.py

```diff
@@ -1342,6 +1345,108 @@ def test_push_bouncer_api(self) -> None:
         self.assertEqual(remote_realm.last_request_datetime, time_sent)
 
 
+class RegisterPushDeviceToBouncer(BouncerTestCase):
+    DEFAULT_SUBDOMAIN = ""
+
+    def get_register_push_device_payload(
+        self,
+        token: str = "test-token",
+        token_kind: int = 2,
+        timestamp: int = datetime_to_timestamp(now()),
+    ) -> dict[str, str | int]:
+        hamlet = self.example_user("hamlet")
+        remote_realm = RemoteRealm.objects.get(uuid=hamlet.realm.uuid)
+
+        push_registration = {
+            "token": token,
+            "token_kind": token_kind,
+            "timestamp": timestamp,
+        }
+
+        assert settings.PUSH_REGISTRATION_ENCRYPTION_KEYS
+        public_key_str: str = next(iter(settings.PUSH_REGISTRATION_ENCRYPTION_KEYS.keys()))
+        public_key = PublicKey(public_key_str.encode("utf-8"), Base64Encoder)
+        sealed_box = SealedBox(public_key)
+        encrypted_push_registration_bytes = sealed_box.encrypt(
+            orjson.dumps(push_registration), Base64Encoder
+        )
+        encrypted_push_registration = encrypted_push_registration_bytes.decode("utf-8")
+
+        payload: dict[str, str | int] = {
+            "user_uuid": str(hamlet.uuid),
+            "realm_uuid": str(remote_realm.uuid),
+            "push_account_id": 2408,
+            "encrypted_push_registration": encrypted_push_registration,
+            "bouncer_public_key": public_key_str,
+        }
+        return payload
+
+    def test_register_push_device_success(self) -> None:
+        remote_push_devices_count = RemotePushDevice.objects.count()
+        self.assertEqual(remote_push_devices_count, 0)
+
+        payload = self.get_register_push_device_payload()
+
+        result = self.uuid_post(
+            self.server_uuid,
+            "/api/v1/remotes/push/e2ee/register",
+            payload,
+        )
+        response_dict = self.assert_json_success(result)
+
+        remote_push_devices = RemotePushDevice.objects.all()
+        self.assert_length(remote_push_devices, 1)
+        self.assertEqual(response_dict["device_id"], remote_push_devices[0].device_id)
+
+        # Idempotent
+        result = self.uuid_post(
+            self.server_uuid,
+            "/api/v1/remotes/push/e2ee/register",
+            payload,
+        )
```

Comment:
You should parse the response here and make sure it returned the right device ID too. 

Actually this test code needs to assert success too -- you never want to make an API call and then not check the result in tests, and it can hide nasty bugs :)